content/blog/jenkinsfile-wishes.md

```diff
@@ -0,0 +1,212 @@
++++
+title = 'Feature Request: Parallel Stage Generation for Declarative Jenkins Pipelines'
+date = 2019-01-26T11:03:06-08:00
+draft = true
+tags = ["tags"]
+description = "Desc"
+
+# For twitter cards, see https://github.com/mtn/cocoa-eh-hugo-theme/wiki/Twitter-cards
+meta_img = "/images/image.jpg"
+
+# For hacker news and lobsters builtin links, see github.com/mtn/cocoa-eh-hugo-theme/wiki/Social-Links
+hacker_news_id = ""
+lobsters_id = ""
++++
+
+# A bit of context
+
+Normally a declarative Jenkins pipeline is pretty simple it can look something
+like this:
+
+```groovy
+pipeline {
+    agent {
+        label "linux&&x86_64"
+    }
+
+    stages {
+        stage("Building / Testing") {
+            stage("Testing") {
+                steps {
+                    sh("make test")
+                }
+            }
+            stage("Building") {
+                steps {
+                    sh("make clean && make")
+                }
+            }
+        }
+    }
+}
+```
+
+But what happens when you need to repeat those stages over and over again,
+accounting for a small difference (in terms of code change) of something like
+CPU architectures?
+
+For our example we'll need to test that our application tests and builds on
+the CPU architectures x86_64, s390x, ppc64le, armv8, armv7, and i386.
+
+# Getting this to work currently
+
+So there's a couple of ways to tackle this currently but both have their
+drawbacks and neither can satisfy the three qualities I like in my pipelines:
+
+1. Fully declarative pipelines (no exec'ing out to `script`)
+2. DRY (if there's a chance to not repeat let's take it)
+3. Maintainability is important
```

Comment:
I think this reads better as the three qualities I like in my pipelines

1. ...
2. ... 
3. Maintainability

Basically saying a quality you like in your pipeline is maintainability.
As opposed to reading as a quality I like in my pipelines is maintainability is important.